src/accountsdb/accounts_file.zig

```diff
@@ -10,7 +10,27 @@ const Pubkey = @import("../core/pubkey.zig").Pubkey;
 
 const AccountFileInfo = @import("snapshots.zig").AccountFileInfo;
 
-pub const FileId = u32;
+/// Simple strictly-typed alias for an integer, used to represent a file ID.
+pub const FileId = enum(u32) {
+    _,
+
+    pub inline fn fromInt(int: u32) FileId {
+        return @enumFromInt(int);
+    }
+
+    pub inline fn toInt(file_id: FileId) u32 {
+        return @intFromEnum(file_id);
+    }
+
+    pub fn format(
+        _: FileId,
+        comptime _: []const u8,
+        _: std.fmt.FormatOptions,
+        _: anytype,
+    ) !void {
+        @compileError("Should not print " ++ @typeName(FileId) ++ " directly");
```

Comment:
why not?

Reply:
I forget which, but I saw a PR in which `FieldId` were being printed as part of a filename somewhere, so I set this up to catch any unintended behavior.

Reply:
makes sense